ledger/participant-integration-api/src/main/scala/platform/store/backend/StorageBackend.scala

```diff
@@ -78,39 +78,43 @@ trait IngestionStorageBackend[DB_BATCH] {
     * @param connection to be used when initializing
     * @return the LedgerEnd, which should be the basis for further indexing.
     */
-  def initialize(connection: Connection): StorageBackend.LedgerEnd
+  def initializeIngestion(connection: Connection): StorageBackend.OptionalLedgerEnd
 }
 
-// TODO append-only: consolidate parameters table facing functions
 trait ParameterStorageBackend {
 
-  /** This method is used to update the parameters table: setting the new observable ledger-end, and other parameters.
+  /** This method is used to update the new observable ledger end.
     * No significant CPU load, mostly blocking JDBC communication with the database backend.
     *
     * @param connection to be used when updating the parameters table
-    * @param params     the parameters
     */
-  def updateParams(params: StorageBackend.Params)(connection: Connection): Unit
+  def updateLedgerEnd(ledgerEnd: StorageBackend.LedgerEnd)(connection: Connection): Unit
 
-  /** Query the ledgerEnd, read from the parameters table.
+  /** Query the ledger end, read from the parameters table.
     * No significant CPU load, mostly blocking JDBC communication with the database backend.
     *
     * @param connection to be used to get the LedgerEnd
     * @return the LedgerEnd, which should be the basis for further indexing
     */
-  def ledgerEnd(connection: Connection): StorageBackend.LedgerEnd
-  def ledgerId(connection: Connection): Option[LedgerId]
-  def updateLedgerId(ledgerId: String)(connection: Connection): Unit
-  def participantId(connection: Connection): Option[ParticipantId]
-  def updateParticipantId(participantId: String)(connection: Connection): Unit
-  def ledgerEndOffset(connection: Connection): Offset
-  def ledgerEndOffsetAndSequentialId(connection: Connection): (Offset, Long)
-  def initialLedgerEnd(connection: Connection): Option[Offset]
+  def ledgerEnd(connection: Connection): StorageBackend.OptionalLedgerEnd
 
   /** Part of pruning process, this needs to be in the same transaction as the other pruning related database operations
     */
   def updatePrunedUptoInclusive(prunedUpToInclusive: Offset)(connection: Connection): Unit
   def prunedUptoInclusive(connection: Connection): Option[Offset]
+
+  /** Initializes the parameters table and verifies or updates ledger identity parameters:
+    *  - If no identity parameters are stored, then they are set to the given value.
```

Comment:
I think it would be sufficient to say that this method is idempotent. (if you prefer to still keep some/all of these details anyway, please add somewhere a line that it is idempotent)